syfertext/vocab.py

```diff
@@ -18,5 +25,178 @@ def __init__(self, model_name: str):
+
+        else:
+            orth = key
+
+        # Get the LexemeMeta object
+        # Note: if it is not present in the lex_store then a new one is created
+        lex_meta = self.lex_store.get(orth)
+
+        return lex_meta is not None
+
+    def has_vector(self, string: str) -> bool:
+        """Check whether the given string has an entry in word vectors table
+        
+        Args:
+            string: The word for which the existence of a vector is to be checked out.
+        
+        """
+
+        return self.vectors.has_vector(string)
+
+    def get_lex_meta(self, orth: int) -> LexemeMeta:
+        """Get a LexemeMeta from the lexstore, creating a new
+        Lexeme if necessary.
+        
+        Args:
+            orth: The word hash for which the LexemeMeta object is requested.
+        """
+
+        if orth == 0:
+            return None
+
+        # get the LexemeMeta object from lex store if exist.
+        lex_meta = self.lex_store.get(orth)
+
+        # If the LexemeMeta object already exist it returned.
+        if lex_meta:
+            return lex_meta
+
+        # LexemeMeta instance doesn't exist for the given orth
+        else:
+            # Create the new LexemeMeta object.
+            return self._create_lex_meta(self.store[orth])
+
+    def _create_lex_meta(self, string: str) -> LexemeMeta:
+        """Creates a LexemeMeta object corresponding to `string`.
+        
+        Args:
+            string: The plaintext string for which a LexemeMeta object is to be created.
+            
+        Returns:
+            A LexemeMeta object corresponding to `string`.
+        """
+
+        # Initialize vectors for the provided language model.
+        # If data is not yet loaded, then load it
+        if not self.vectors.loaded:
+            self.vectors._load_data()
+
+        # create the new LexemeMeta object
+        lex_meta = LexemeMeta()
+
+        # Assign the lex attributes
+        lex_meta.orth = self.store.add(string)
+        lex_meta.length = len(string)
+
+        # The language model name of parent vocabulary
+        lex_meta.lang = self.store.add(self.model_name)
+
+        # id is the index of the corresponding vector
+        # in self.vectors
+        lex_meta.id = self.vectors.key2row.get(lex_meta.orth)
+
+        # Traverse all the lexical attributes getters in the dict.
+        for attr, func in self.lex_attr_getters.items():
+            value = func(string)
+
+            # check if string id out of vocabulary
+            if attr == Attributes.IS_OOV:
+                value = not self.has_vector(string)
+
+            # For attributes with string values add them to string store
+            # and assign the orth id of that to LexemeMeta object
+            if isinstance(value, str):
+                value = self.store.add(value)
+
+            # Assign rest of the attributes to the LexemeMeta object
+            if value:
+                Lexeme.set_lex_attr(lex_meta, attr, value)
```

Comment:
So, the idea is LexemeMeta class is not to be directly used by a user, and user can directly set new attributes using this method of lexeme class, LexemeMeta class is to be used for internal funiculitis only like TokenMeta. User should be exposed to Lexeme class only.

Reply:
The issue is that the user of this method still needs access to the `LexemeMeta` object `lex_meta`. So you are using `Lexeme` to set an attribute to a `LexemeMeta` object, I suggest to simple make this as a method of LexemeMeta. `lex_meta.set_attr(attr, value)` Unless you have some fundamental objections

Reply:
yeah you are right. will do

Reply:
This is not the behaviour I am expecting, want to keep this method within LexemeMeta, but needs to remove the lex_meta object somehow, let me think this might take some time. 
i mean a user should not have to access LexemeMeta objects directly

Reply:
The user would still not need to have a direct access to LexemeMeta. The set_attr method I am suggesting would not be used by the end user.